fe/fe-core/src/main/java/com/starrocks/scheduler/TaskManager.java

```diff
@@ -45,19 +52,21 @@ public class TaskManager {
 
     private static final Logger LOG = LogManager.getLogger(TaskManager.class);
 
-    public static final long TASK_EXISTS = -1L;
-    public static final long DUPLICATE_CREATE_TASK = -2L;
-    public static final long GET_TASK_LOCK_FAILED = -3L;
-
-    // taskId -> Task , Task may have Manual Task, Periodical Task
+    // taskId -> Task , Task may have Normal Task, Periodical Task
     // every TaskRun must be generated by a Task
-    private final Map<Long, Task> manualTaskMap;
+    private final Map<Long, Task> normalTaskMap;
+    private final Map<Long, Task> periodTaskMap;
+    private final Map<Long, ScheduledFuture<?>> periodFutureMap;
     // taskName -> Task, include Manual Task, Periodical Task
     private final Map<String, Task> nameToTaskMap;
```

Comment:
Why need the id map and name map at the same time?

Reply:
Because the user is adding tasks or deleting tasks through Name, the index of Name is required. When we do operations, we usually use IDs to prevent Task renaming or some other reason.

Reply:
Is it necessary to use 2 maps? If there is new type of task, will you use 3 maps to save them separately? I think we can just use one map idToTaskMap to save all the tasks. If you need to operate on all tasks separately, you can distinguish them by type.